lib/rubocop/cop/mixin/hash_shorthand_syntax.rb

```diff
@@ -46,10 +46,22 @@ def require_hash_value?(hash_key_source, node)
 
       def without_parentheses_call_expr_follows?(node)
         return false unless (ancestor = node.parent.parent)
-        return false unless (right_sibling = ancestor.right_sibling)
 
-        ancestor.respond_to?(:parenthesized?) && !ancestor.parenthesized? &&
-          right_sibling.respond_to?(:parenthesized?) && !right_sibling.parenthesized?
+        right_sibling = ancestor.right_sibling
+
+        return true if right_sibling.nil? && not_parenthesized_node?(ancestor)
+        return false unless right_sibling
+        return true if node_with_block_and_arguments?(right_sibling)
+
+        not_parenthesized_node?(ancestor) && not_parenthesized_node?(right_sibling)
+      end
+
+      def not_parenthesized_node?(node)
```

Comment:
How about method name like the following?

```suggestion
      def without_parentheses?(node)
```